crypto/engine/eng_dyn.c

```diff
@@ -484,7 +484,9 @@ static int dynamic_load(ENGINE *e, dynamic_data_ctx *ctx)
     engine_set_all_null(e);
 
     /* Try to bind the ENGINE onto our own ENGINE structure */
-    if (!ctx->bind_engine(e, ctx->engine_id, &fns)) {
+    if (!engine_add_dynamic_id(e, (ENGINE_DYNAMIC_ID)ctx->bind_engine, 1)
+            || !ctx->bind_engine(e, ctx->engine_id, &fns)) {
+        engine_remove_dynamic_id(e, 1);
```

Comment:
Wouldn't this mean that a double-loaded engine will also remove the dynamic_id of the original engine instance? I agree that the engine won't be loaded twice (bind_engine() won't be called), but if the load will happen 3 times won't the engine be loaded twice?

Reply:
nvm, I saw what you did the e->dynamic_id, and setting it up only after everything is good. Nice solution.